resources/deployments/controller/controller_deployments.go

```diff
@@ -341,6 +369,18 @@ func (d *DeploymentsController) LookupDeployment(w rest.ResponseWriter, r *rest.
 		return
 	}
 
+	for idx, dep := range deps {
+		//REVIEW: should I return here on error?
+
+		deviceDeployments, err := d.model.GetDeviceStatusesForDeployment(ctx, *dep.Id)
+		if err != nil {
+			d.view.RenderError(w, r, err, http.StatusBadRequest, l)
+			return
+		}
+
+		deps[idx].Count = len(deviceDeployments)
```

Comment:
maybe there is a better way of counting objects then retrieving all of them (there is a count in mongo, but I don't know how it works)

Reply:
can give this a shot:

https://docs.mongodb.com/manual/reference/method/db.collection.count/

Reply:
Don't count it ourself, use count query.

Reply:
Alternatively, cache the count then creating deployment object in the first place. This way you avoid having to extra query